be/src/exprs/function_context.cpp

```diff
@@ -190,4 +195,67 @@ const FunctionContext::TypeDesc* FunctionContext::get_arg_type(int arg_idx) cons
     return &_arg_types[arg_idx];
 }
 
+struct ColumnBuilder {
+    template <LogicalType Type>
+    ColumnPtr operator()(const FunctionContext::TypeDesc& type_desc) {
+        if constexpr (lt_is_decimal<Type>) {
+            return RunTimeColumnType<Type>::create(type_desc.precision, type_desc.scale);
+        } else if constexpr (lt_is_conllection<Type>) {
+            throw std::runtime_error(fmt::format("Unsupported collection type {}", Type));
+            return nullptr;
+        } else if constexpr (Type == TYPE_UNKNOWN || Type == TYPE_BINARY || Type == TYPE_DECIMAL) {
+            throw std::runtime_error(fmt::format("Unsupported column type {}", Type));
+            return nullptr;
+        } else {
+            return RunTimeColumnType<Type>::create();
+        }
+    }
+};
+
+ColumnPtr FunctionContext::create_column(const FunctionContext::TypeDesc& type_desc, bool nullable) {
+    const auto type = type_desc.type;
```

Comment:
columnHelper has similar functions, can they be merged together?
```
ColumnPtr ColumnHelper::create_column(const TypeDescriptor& type_desc, bool nullable, bool is_const, size_t size,
                                      bool use_adaptive_nullable_column) {
```

Reply:
the inferface is different